subscriber/common/job/controller.go

```diff
@@ -129,6 +134,17 @@ func NewController(params Params) *Controller {
 
 	if params.ServiceConfig.ControllerConfig.Enable {
 		params.ServiceConfig.Logger.Info("aresDB Controller is enabled")
+
+		if *params.ServiceConfig.HeartbeatConfig.Enabled {
+			controller.etcdServices, err = createEtcdServices(params)
```

Comment:
connectEtcdService?

Reply:
1. params.ServiceConfig.HeartbeatConfig.Enabled is a temporary variable to differentiate zk and etcd. Eventually, it will be removed once zk is gone.
2. connectEtcdService is used to create etcd services client.